CMakeLists.txt

```diff
@@ -258,11 +254,31 @@ if(NOT APPLE OR NOT ${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64")
   set(IREE_HAL_DRIVER_METAL_DEFAULT OFF)
 endif()
 
+# HIP support can be enabled on any platform.
+set(IREE_HAL_DRIVER_HIP_DEFAULT ${IREE_HAL_DRIVER_DEFAULTS})
+if(CMAKE_CROSSCOMPILING)
+  set(IREE_HAL_DRIVER_HIP_DEFAULT OFF)
+endif()
+
 option(IREE_HAL_DRIVER_CUDA "Enables the 'cuda' runtime HAL driver" ${IREE_HAL_DRIVER_CUDA_DEFAULT})
 option(IREE_HAL_DRIVER_LOCAL_SYNC "Enables the 'local-sync' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
 option(IREE_HAL_DRIVER_LOCAL_TASK "Enables the 'local-task' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
 option(IREE_HAL_DRIVER_VULKAN "Enables the 'vulkan' runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
 option(IREE_HAL_DRIVER_METAL "Enables the 'metal' runtime HAL driver" ${IREE_HAL_DRIVER_METAL_DEFAULT})
+option(IREE_HAL_DRIVER_HIP "Enables the 'hip' runtime HAL driver" ${IREE_HAL_DRIVER_HIP_DEFAULT})
```

Comment:
> Refactors the IREE_EXTERNAL_HAL_DRIVERS facility to source from implicitly enabled drivers and user specified (the hip driver is the first to need this distinction).

If we're adding an explicit option here, do we even need to use `IREE_EXTERNAL_HAL_DRIVERS` for HIP? Maybe we should add the option at the same time as when we move https://github.com/openxla/iree/tree/main/experimental/hip out of experimental?

Might be fine to set things up this way for a transition period... but we might be able to avoid the complexity.

(+1 to Ben's comment)